### PR TITLE
Spell checker

### PR DESCRIPTION
**Pull request by ahmadshadab**

### CODE REVIEW

**Review by eddturner — CHANGES_REQUESTED** [review]
There might be a change required in one of the tests

File: common-rest/src/main/java/org/uniprot/api/common/repository/search/SolrRequestConverter.java [hunk 1]
```diff
@@ -66,8 +66,13 @@ public JsonQueryRequest toJsonQueryRequest(SolrRequest request, boolean isEntry)
...
 
-        solrQuery.add("spellcheck", "true");
-
+        //        solrQuery.add("spellcheck", "true");
+        //        solrQuery.add("spellcheck.count", "5");
+        //        solrQuery.add("spellcheck.maxResultsForSuggest", "0");
+        //        solrQuery.add("spellcheck.collate", "true");
+        //        solrQuery.add("spellcheck.collateExtendedResults", "true");
+        //        solrQuery.add("spellcheck.maxCollations", "5");
+        //        solrQuery.add("spellcheck.maxCollationTries", "5");
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply common-rest/src/main/java/org/uniprot/api/common/repository/search/SolrRequestConverter.java]: done

File: uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBSearchControllerIT.java [hunk 2]
```diff
@@ -1260,6 +1282,95 @@ void searchWithCrossRefTCDBValue() throws Exception {
...
+                        .perform(
+                                get(SEARCH_RESOURCE + "?query=cambridge")
+                                        .header(ACCEPT, APPLICATION_JSON_VALUE));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.results", is(empty())))
+                .andExpect(jsonPath("$.suggestions").doesNotExist());
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply uniprotkb-rest/src/test/java/org/uniprot/api/uniprotkb/controller/UniProtKBSearchControllerIT.java:1371]: Good point. I am not sure, what is the right thing to do. Lets talk about it.

File: help-centre-rest/src/test/java/org/uniprot/api/help/centre/controller/HelpCentreSearchControllerIT.java [hunk 3]
```diff
@@ -240,23 +251,25 @@ void singleSuggestionGivenForMultiWordQuery() throws Exception {
         ResultActions response =
                 mockMvc.perform(
                         get(getSearchRequestPath())
-                                .param("query", "protein bell")
+                                .param("query", "\"protein bell\"")
```
[the remark this answers is not in the document]
Comment (ahmadshadab) [reply help-centre-rest/src/test/java/org/uniprot/api/help/centre/controller/HelpCentreSearchControllerIT.java:254]: Good catch. When i search a free text without quote, the suggestion adds default operator AND for the space. So the suggestions becomes `protein AND bell`. Can we talk about it?

**Pull request merged**